### ISSUE TITLE
Please do not update global state in the constructor of cli commands!

### ISSUE THREAD

**Issue by hostep** [magmodules/magento2-channable#22]
Hi guys

This was encountered with version 1.0.8 of your Channable module installed.

When running `bin/magento setup:store-config:set --base-url="https://blah.example/"`
We get this error:
```
[Magento\Framework\Exception\LocalizedException]
  Area code is already set
```

This problem is triggered by the `setAreaCode` code in:
- https://github.com/magmodules/magento2-channable/blob/1.0.8/Console/Command/Config.php#L58
- https://github.com/magmodules/magento2-channable/blob/1.0.8/Console/Command/ItemUpdate.php#L67

Explanation: every time you run `bin/magento` it executes **all** constructors of all registered cli commands. So this means all the constructors of your `channable:***` cli commands get executed even when executing another non-related `bin/magento` command.
You can very easily test this by echoing something from the constructors and running whatever `bin/magento` command, and you will always see the output from your commands in every other command.

So you should only use the constructor of these commands to initialize a bunch of variables, but please don't execute logic, and certainly not logic which manipulates a global state (like setting the area code).

The correct way, would be to change the area code in the `execute` method, but certainly not in the constructor.

This is a really big problem and should be resolved quickly in my opinion.

Thanks! :)

**Comment by hostep** [comment]
Please review your other Magento 2 modules for similar problems, I can see that in for example your 'magento2-sooqr' module, the same mistake was made.

**Comment by magmodules** [comment]
Hi, we are working on an fix at this moment. The problem is a bit different, as this is causes by an session being set somewhere down the line. Moving this to the execute won't solve the problem at this moment. We will issue a new version later today

**Comment by magmodules** [comment]
Fixed in 1.0.9, https://github.com/magmodules/magento2-channable/releases/tag/1.0.9

**Comment by hostep** [comment]
Thanks!
